marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/BlockInfo.hs

```diff
@@ -88,6 +122,53 @@ propTrippingBlockInfoJSON = Hedgehog.property $ do
   event <- Hedgehog.forAll genBlockInfo
   Hedgehog.tripping event Aeson.encode Aeson.eitherDecode
 
+-- | Test for block info using cardano-node-emulator
+endToEndBlockInfo :: Hedgehog.Property
+endToEndBlockInfo = Helpers.unitTestWithTmpDir "." $ \tempPath -> do
+  -- Setup
+  (trace, _) <- liftIO $ defaultStdOutLogger "endToEndBlockInfo"
+  let marconiTrace = mkMarconiTrace trace
+
+  -- Local node config and connect info, with slots of length 100ms
+  (nscConfig, _) <- Hedgehog.evalIO $ Integration.mkLocalNodeInfo tempPath 100
+
+  -- Indexer preprocessor and configuration
+  let
+    catchupConfig = Integration.mkEndToEndCatchupConfig
+    config = Integration.mkEndToEndRunIndexerConfig marconiTrace nscConfig Runner.withDistancePreprocessor
+
+  StandardWorker mindexer worker <-
+    Hedgehog.evalIO $
+      either throwIO pure
+        =<< runExceptT
+          (blockInfoBuilder (config ^. Runner.runIndexerConfigSecurityParam) catchupConfig trace tempPath)
+  coordinator <- Hedgehog.evalIO $ Core.mkCoordinator [worker]
+
+  -- Start the testnet and indexer, and run the query.
+
+  {- NOTE: PLT-8098
+   startTestnet does not shutdown when the test is done.
+   As a temporary measure to avoid polluting the test output, Integration.startTestnet squashes all
+   SlotAdd log messages.
+   -}
+  Hedgehog.evalIO $ Integration.startTestnet nscConfig
+
+  res <- Hedgehog.evalIO $
+    Async.race (Runner.runIndexer config coordinator) $
+      do
+        threadDelay 5_000_000
```

Comment:
Do we need an additionnal story for retrying if 5s is not enough?

Reply:
Good point. I think it is appropriate to make it part of the additional story to improve how transactions are handled, since in effect we are waiting until the transaction is submitted to the local network.

I will add it there and we can discuss on the ticket if you would like to see the retry mechanism as a separate story.